clock/testing/fake_clock.go

```diff
@@ -231,6 +249,12 @@ func (*IntervalClock) Tick(d time.Duration) <-chan time.Time {
 	panic("IntervalClock doesn't implement Tick")
 }
 
+// NewTicker is currently unimplemented, will panic.
+// TODO: make interval clock use FakeClock so this can be implemented.
+func (*IntervalClock) NewTicker(d time.Duration) clock.Ticker {
```

Comment:
Why do we need this function again? If we didn't, IntervalClock wouldn't implement the "WithTicker" interface, which means that it would fail at compile time. By adding it, it now implements WithTicker, but will fail at runtime?

Reply:
True.  BTW, IntervalClock also has a lot of unimplemented methods for `Clock`.

Reply:
We could remove them all, leaving IntervalClock implementing `PassiveClock` --- which would be a compile-time tight characterization.

Reply:
There are no uses of IntervalClock in k8s.io/utils.

In the main repo there are only two files that use it: `client-go/tools/record/event_test.go` and `client-go/tools/record/events_cache_test.go`.

Reply:
Let's make it possible to remove the api machinery clock. We should also not implement interfaces that don't actually work -- this library is for public consumption and that will be very annoying to a consumer.

Reply:
Ugh, there is non-trivial entanglement with code that needs a non-passive clock.  In `client-go/tools/record/events_cache.go`, the Filter method of EventSourceObjectSpamFilter passes its clock to `flowcontrol.NewTokenBucketRateLimiterWithClock` --- which needs a clock that can `Sleep`.  Why does this work?

Reply:
@lavalamp : I think #212 does what you said.